hist/hist/src/TGraph2D.cxx

```diff
@@ -1410,6 +1412,34 @@ TH1 *TGraph2D::Project(Option_t *option) const
 }
 
 
+////////////////////////////////////////////////////////////////////////////////
+/// Deletes duplicated points.
+///
+/// The Delaunay triangulation algorithm assumes that each (x, y) coordinate corresponds to a unique z value,
+/// meaning duplicate (x, y) points are not allowed. Consequently, when using drawing options that rely on this
+/// algorithm (e.g., TRI, SURF, etc.), a warning may appear instructing you to remove duplicates.
+/// This function provides a way to handle such duplicates.
+///
+/// Example:
+/// ~~~ {.cpp}
+/// g->RemoveDuplicates();
+/// g->Draw("TRI1");
+/// ~~~
+
+Int_t TGraph2D::RemoveDuplicates()
+{
+   for (int i=0; i<fNpoints; i++) {
+      double x = fX[i];
+      double y = fY[i];
+      for (int j=i+1; j<fNpoints; j++) {
+         if (x==fX[j] && y==fY[j]) {RemovePoint(j); j--;}
```

Comment:
Won't this cause 3 deallocations, 3 allocations and 3 copies per point removed?
Can we:
- Avoid allocations and deallocations in `RemovePoint` by simply decrementing `fNPoints`?
- Add a private method, `RemovePoints`, that removes points in bulk, iterating on the indices to remove? `RemovePoint` can invoke the bulk method internally for 1 point.

Reply:
Ok, I will investigate the way `RemovePoint` is coded.